src/main/java/com/dev/boot/amazonreviewsapp/service/impl/ReviewServiceImpl.java

```diff
@@ -12,38 +18,44 @@
     @Autowired
     private ReviewRepository reviewRepository;
 
-    @Override
-    public Review add(Review review) {
-        return reviewRepository.save(review);
-    }
-
     @Override
     public List<Review> addAll(List<Review> reviews) {
         return reviewRepository.saveAll(reviews);
     }
 
     @Override
-    public List<Review> getByUser(String userId) {
-        return reviewRepository.findAll();
+    public Set<String> getMostActiveUsers() {
+        return reviewRepository.getMostActiveUsers();
     }
 
     @Override
-    public List<Review> getByProduct(String productId) {
-        return reviewRepository.findAll();
+    public Set<String> getMostPopularProducts() {
+        return reviewRepository.getMostPopularProducts();
     }
 
     @Override
-    public List<String> getMostActiveUsers() {
-        return null;
+    public Set<String> getMostPopularWordsInReviews() {
+        Map<String, Integer> words = reviewRepository.getReviewsText()
+                .stream()
+                .flatMap(text -> Arrays.stream(text.replaceAll("[^a-zA-Z\\s]", "").split(" ")))
+                .collect(Collectors.toMap(k -> k, k -> 1, (a, b) -> a + b, HashMap::new));
+
+        return words.values()
+                .stream()
+                .distinct()
+                .flatMap(val -> getKeys(words, val).stream())
```

Comment:
do we need to sort words?

Reply:
we sort not words, but their frequency

Reply:
this is mistake(
distinkt remove repeated values...

Reply:
fixed!